src/mono/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Mono.cs

```diff
@@ -193,48 +195,56 @@ public static int GetExceptionCode()
             throw new PlatformNotSupportedException();
         }
 
+        [RequiresDynamicCode("Marshalling code for the object might not be available")]
```

Comment:
Hmm, not sure about these. Does this add any value?

Reply:
The `APICompat` check fails if a public implementation API doesn't match the reference API (and vice versa). 

The `CoreRT `implementation required this API to be annotated, that in turn required the reference API to be annotated which is shared by all implementations, which required other runtime implementations to be annotated.

Reply:
Yeah, these APIs are already marked obsolete and there will be _some_ warning for them. But the AOT warnings generally follow the philosophy of trimming warnings: if there's a possible behavioral difference, there should be a warning.

These APIs allow running marshalling on anything, then read a byte out the marshalled representation, and destroy it. They can't work if we can't build the marshaller dynamically at runtime (and it wasn't precompiled).